server/lib/tuist_web/controllers/api/bundles_controller.ex

```
@@ -20,6 +22,170 @@ defmodule TuistWeb.API.BundlesController do
+               properties: %{
+                 current_page: %Schema{type: :integer},
+                 page_size: %Schema{type: :integer},
+                 total_count: %Schema{type: :integer},
+                 total_pages: %Schema{type: :integer}
+               },
+               required: [:current_page, :page_size, :total_count, :total_pages]
+             }
+           },
+           required: [:data, :meta]
+         }},
+      unauthorized: {"You need to be authenticated to list bundles", "application/json", Error},
+      forbidden: {"You are not authorized to list bundles", "application/json", Error}
+    }
+
+  def index(%{assigns: %{selected_project: selected_project}} = conn, params) do
+    # Create filters in the format Flop expects (list of filter objects)
+    filters = [
+      %{field: :project_id, op: :==, value: selected_project.id}
+    ]
+
+    filters =
+      case Map.get(params, "git_branch") do
+        nil -> filters
+        branch -> [%{field: :git_branch, op: :==, value: branch} | filters]
+      end
+
+    flop_params = %{
+      filters: filters,
+      order_by: [:inserted_at],
+      order_directions: [:desc],
+      page_size: Map.get(params, "page_size", 50)
+    }
+
+    flop_params =
+      case Map.get(params, "page") do
+        nil -> flop_params
+        page -> Map.put(flop_params, :page, page)
+      end
+
+    case Bundles.list_bundles(flop_params, preload: [:uploaded_by_account, project: :account]) do
+      {bundles, meta} ->
+        conn
+        |> put_status(:ok)
+        |> json(%{
+          data: Enum.map(bundles, &bundle_list_to_map/1),
+          meta: %{
+            current_page: meta.current_page,
+            page_size: meta.page_size,
+            total_count: meta.total_count,
+            total_pages: meta.total_pages
+          }
+        })
+
+      error ->
+        conn
+        |> put_status(:internal_server_error)
+        |> json(%{error: "Failed to list bundles: #{inspect(error)}"})
```

the client doesn't read the error. I'm not sure if I would be explicitly handling an internal server error, I'd just let the process blow up. If you do it like this, we don't get the data in app signal which is certainly not ideal.

---

I'll remove it.